### DISCUSSION TITLE
coordinates of mesh vertices

### DISCUSSION THREAD

**Discussion by jcddafonseca** [firedrakeproject/firedrake#3131] [Firedrake support]
Hi!
I'm a newcomer to finite element methods and Firedrake. 
Based on Firedrake documentation, I'm trying to answer the following question:
How could I get the coordinate values of the vertices belonging to an instance of the  class Mesh?
Would there be a simple way to do that using a specific class method or function?

Many thanks!
Júlio Fonseca

**Comment by connorjward** [comment]
You can access the mesh coordinate array using `mesh.coordinates.dat.data` ([docs](https://www.firedrakeproject.org/mesh-coordinates.html)). Note that we renumber the mesh internally so these values might not be in a "sensible" order.

**Reply by jcddafonseca** [reply]
Thanks Connor! This is I was looking for...
